### PR TITLE
Prepare for public preview release

### PR DESCRIPTION
**Pull request by hsubox76**
- Rename package (@firebase/) and top level entry point (firebase/) from "vertexai" to "vertexai-preview".
- Fix any missed renames from "vertex" to "vertexAI"
- Add NPM token for the new package to the 3 workflows that do NPM publishes (already added to Github secrets)
- Add `api-report` npm script so that it will be picked up by the docgen process (will generate docs in a separate PR)
- Fix comments flagged by api-documenter process
- Add changeset (set vertexai-preview version to 0.0.0 and allow changesets to patch bump it to 0.0.1 on publish). Note the changeset will apply to the whole `ch-vertex-feature` branch being merged into `master`, so that's why it includes `@firebase/app`.

I did not rename the directory `packages/vertexai` because I didn't want this PR to have a 140 file diff. It should be fine, the dirname isn't published to NPM.

### CODE REVIEW

File: packages/vertexai/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@firebase/vertexai",
-  "version": "0.0.1",
-  "description": "A template package for new firebase packages",
+  "name": "@firebase/vertexai-preview",
+  "version": "0.0.0",
+  "description": "A Firebase SDK for VertexAI",
```
Comment (DellaBitta) [inline packages/vertexai/package.json:4]: Suggestion: maybe we should we note that it's a preview version here in the description?

File: packages/vertexai/src/api.ts [hunk 2]
```diff
@@ -45,20 +45,26 @@ declare module '@firebase/component' {
 export function getVertexAI(app: FirebaseApp = getApp()): VertexAI {
   app = getModularInstance(app);
   // Dependencies
-  const vertexProvider: Provider<'vertex'> = _getProvider(app, VERTEX_TYPE);
+  const vertexProvider: Provider<'vertexAI'> = _getProvider(app, VERTEX_TYPE);
```
Comment (DellaBitta) [inline packages/vertexai/src/api.ts:48]: Could change Provider<\`vertexAI\`> to Provider<VERTEX_TYPE> here, I think.

File: packages/vertexai/src/types/responses.ts [hunk 3]
```diff
@@ -74,6 +74,11 @@ export interface GenerateContentResponse {
   usageMetadata?: UsageMetadata;
 }
 
+/**
+ * Usage metadata about response.
```
Comment (DellaBitta) [inline packages/vertexai/src/types/responses.ts:78]: nit `a response`

**Review by dlarocque — APPROVED** [review]

File: packages/vertexai/package.json (hunk 1 above)
Comment (hsubox76) [reply packages/vertexai/package.json:4]: Added, hopefully we catch this string when we go GA!

File: packages/vertexai/src/api.ts (hunk 2 above)
Comment (hsubox76) [reply packages/vertexai/src/api.ts:48]: Sadly I cannot, Typescript can't turn an actual code string into a string that you use for typing.

File: packages/vertexai/src/types/responses.ts (hunk 3 above)
Comment (hsubox76) [reply packages/vertexai/src/types/responses.ts]: Could even link to `GenerateContentResponse`, will do that.

**Pull request merged**